src/pybind/mgr/dashboard/frontend/src/app/shared/directives/permission-scope.directive.ts

```diff
@@ -0,0 +1,10 @@
+import { Directive, Input } from '@angular/core';
```

Comment:
I don't like having a dummy directive so I investigated a bit how to improve it and I think  I have a nice solution:
If we change this directive to the code below, we can remove UpdateScopeDirective  and use only this directive.
We can use the disable() method from FormGroup, which is already associated with each form element, instead of manually changing the attribute for each element.

We should also improve the name of directive and attribute, to better relate one with the other.

````
import { Directive, Input, OnInit } from '@angular/core';

import { CdFormGroup } from '../forms/cd-form-group';
import { Permissions } from '../models/permissions';
import { AuthStorageService } from '../services/auth-storage.service';

@Directive({
  selector: '[cdScope]'
})
export class PermissionScopeDirective implements OnInit {
  @Input() cdScope: keyof Permissions;
  @Input() formGroup: CdFormGroup;

  constructor(private authStorageService: AuthStorageService) {}

  ngOnInit() {
    const permissions = this.authStorageService.getPermissions();

    if (!permissions[this.cdScope]?.update) {
      this.formGroup.disable();
    }
  }
}
```

Reply:
Interesting proposal @tspmelo !

My suggestion for the (still trivial) Scope directive is to not be limited to this use-case but to resolve also the permission-based conditional displaying of elements with it: now we need to inject AuthService in every component and then perform `ngIf` directives for every specific HTML template chunk.

By using a declarative directive, we might simply include the hide/display logic in the Scope directive, so we decouple the component logic from the scope/permission display/hide concerns.

Reply:
I could only find a few examples with that situations, like `osd-pg-scrub-modal.component.html`, and the elements that are removed with `ngIf` check a different scope than the one used on the form, In this case the form uses `osd` but the button uses `configOpt`, so we could never read that from the parent directive.

I think the best solution for this problem would be to have new structural directive to remove the elements based on a permission. That way we could remove the injection from the component. But that is another PR.

Reply:
Mmm, nope, maybe I didn't explain myself properly, but this behaviour happens dozens of times accross the code. Just to give you a more specific idea on what I was thinking of:
- In a WiP PR of mine, for the Navigation menu, I needed a similar (structural) directive [`FeatureDirective`](https://github.com/ceph/ceph/pull/33375/files#diff-39edd13f6fefb7dd78cf6943ffe0a40e). As the use case is the same as this one here, I suggested @nizamial09 to modify this PR in order to align both of them and have a reusable approach for permission-based hiding and features/scope tagging.
- Most components that aggregate information from multiple components have multiple `ngIf` directives (the already mentioned Navigation, the Health component, all Grafana iframes...).
- Back-end endpoints that aggregate info from different sources (e.g.: health, summary, ...) implicitly perform the permission checking and hide fields accordingly (I find this quite tricky, as the schema of every API response may change depending on fine-grained permissions).

My proposal for using a high-level Scope directive + another nested Directive (this UpdatePermissions) was to decouple that part of the logic from every specific component and have a generic approach. There were 2 possible implementations:
1. Smart higher-level Scope directive + dummy lower-level ones, and use ViewChildren queries to access lower levels (unfortunately regular HTML elements cannot be retrieved by this way, so the only way is using Directives selecting those). 
2. Dummy higher-level Scope directive + smart lower-level one with DI.

I also suggested Nizam to use generic selectors (same as what ngForm, RouterLink or other Angular built-in directives do) in order to minimize the amount of code changed.

Reply:
@tspmelo WDYT?